host-select: fix compatibility with force-condemned hosts (#6623)
cylc/cylc-flow: changes.d/6623.fix.md

> Auto restart: The "force condemn" option (that tells workflows running on a
> server to shutdown as opposed to migrate) hasn't worked with the host-selection
> mechanism since Cylc 8.0.0. This has now been fixed and the "force condemn"
> option has been restored in the documentation.

Reviewer: ```suggestion
Restored the option to shut down rather than auto-migrate workflows on a condemned
host. This option hasn't worked with the host-selection mechanism since Cylc 8.0.0. 
```

Author: This isn't actually correct.
